### PR TITLE
MM-60570 - Server-side changes consolidating the export CLI with server/ent code

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Add `UntilUpdateAt` for `MessageExport` and `AnalyticsPostCount` -- this is needed for bounded exports (e.g., in the CLI and possibly in the future through the system console)
- Removed quite a few non-UI facing i18n strings -- slowly removing AppError from the internal code
- Add `TranslationsPreInitFromBuffer` to allow CLI to use i18n

Linked: https://github.com/mattermost/enterprise/pull/1769
- https://github.com/mattermost/mattermost-compliance-export/pull/16

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-60570

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/public/shared/i18n/i18n.go [hunk 1]
```diff
@@ -78,6 +78,28 @@ func TranslationsPreInit(translationsDir string) error {
 	return initTranslationsWithDir(translationsDir)
 }
 
+// TranslationsPreInitFromBuffer loads translations from a buffer -- useful if
+// we need to initialize i18n from an embedded i18n file (e.g., from a CLI tool)
+func TranslationsPreInitFromBuffer(filename string, buf []byte) error {
```
Comment (cpoile) [inline server/public/shared/i18n/i18n.go:83]: This is unfortunate, but the problem is: the CLI uses enterprise code which uses a couple translations to output feedback to the system console UI. So the feedback needs to be translated.
For now, the CLI only needs to support english, so this function is enough to translate only those couple of lines.
Eventually we're going to move the CLI's functionality into mmctl, and then we won't need this code anymore. I've added https://mattermost.atlassian.net/browse/MM-60972 to track removing this.

**Review by agnivade — APPROVED** [review]

**Review by lieut-data — APPROVED** [review]
Looks good! Nothing blocking on my side :)

File: server/channels/store/sqlstore/compliance_store.go [hunk 2]
```diff
@@ -323,8 +323,13 @@ func (s SqlComplianceStore) MessageExport(c request.CTX, cursor model.MessageExp
...
 		OrderBy("PostUpdateAt, PostId").
-		Limit(uint64(limit)).
-		ToSql()
+		Limit(uint64(limit))
+
+	if cursor.UntilUpdateAt > 0 {
+		builder = builder.Where(sq.LtOrEq{"Posts.UpdateAt": cursor.UntilUpdateAt})
+	}
+
+	query, args, err := builder.ToSql()
```
[the remark this answers is not in the document]
Comment (cpoile) [reply server/channels/store/sqlstore/compliance_store.go:332]: Agreed!  [Maybe a ticket to improve?  ](https://mattermost.atlassian.net/browse/MM-60989)

File: server/channels/store/sqlstore/post_store.go [hunk 3]
```diff
@@ -2383,6 +2383,10 @@ func (s *SqlPostStore) AnalyticsPostCount(options *model.PostCountOptions) (int6
 		})
 	}
 
+	if options.UntilUpdateAt > 0 {
+		query = query.Where(sq.LtOrEq{"p.UpdateAt": options.UntilUpdateAt})
+	}
+
 	queryString, args, err := query.ToSql()
```
[the remark this answers is not in the document]
Comment (cpoile) [reply server/channels/store/sqlstore/post_store.go]: Yep, sounds good.

**Pull request merged**